### ISSUE TITLE
Retry Managed Identity requests on 410 responses 

### ISSUE THREAD

**Issue by fbreitwieser** [Azure/azure-storage-azcopy#2486]
### Which version of the AzCopy was used? 
10.21.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

### What command did you run?
`azcopy rm` and `azcopy cp` with `AZCOPY_AUTO_LOGIN_TYPE=MSI` and `AZCOPY_MSI_CLIENT_ID` set.

### What problem was encountered?
We run dozens of azcopy commands in parallel in an analysis pipeline. Sometimes, the azcopy commands fail with a 410 error, which sometimes kills the pipelines and requires their restart. 

Full error message:
```
RESPONSE 410 Gone
--------------------------------------------------------------------------------
<?xml version="1.0" encoding="utf-8"?>
<Error xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <Code>ResourceNotAvailable</Code>
    <Message>The resource requested is no longer available. Please refresh your cache.</Message>
    <Details></Details>
</Error>
--------------------------------------------------------------------------------
To troubleshoot, visit https://aka.ms/azsdk/go/identity/troubleshoot#managed-id.
```

### How can we reproduce the problem in the simplest way?
Happens stochastically.

### Have you found a mitigation/solution?
No - need to restart the azcopy commands. On restart the request finishes usually successfully.

### Related
This is likely related to the issue and fix in azure-sdk-for-net in https://github.com/Azure/azure-sdk-for-net/issues/28568 and in azure-sdk-for-python in https://github.com/Azure/azure-sdk-for-python/pull/32200 .

Made easier to implement with #2437 .

**Comment by gapra-msft** [comment]
@chlowell this seems like an issue that should go to the azure-sdk-for-go identity team. Looks like this has been fixed in .net and python already.

**Comment by chlowell** [comment]
This was fixed by https://github.com/Azure/azure-sdk-for-go/pull/21628 which shipped last month in azidentity v1.4.0. I see AzCopy 10.22.0 requires that version, so please try upgrading AzCopy.

**Comment by fbreitwieser** [comment]
Thanks a lot @chlowell and @gapra-msft , will try with the upgraded version! I'll reopen if the issue persisted
